app/src/main/java/com/epfl/drawyourpath/pathDrawing/PathDrawingModel.kt

```diff
@@ -0,0 +1,122 @@
+package com.epfl.drawyourpath.pathDrawing
+
+import android.os.Handler
+import android.os.Looper
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.map
+import com.epfl.drawyourpath.path.Path
+import com.epfl.drawyourpath.path.Run
+import com.google.android.gms.maps.model.LatLng
+import java.time.Duration
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.temporal.ChronoUnit
+
+class PathDrawingModel : ViewModel() {
+
+    private var resultingRun = Run(Path(), 0, 1)

```

Comment:
Maybe declare the resultingRu. as a latent var instead of giving a useless initaial value. 
private lateinit var resultingRun: Run

Reply:
It is there to avoid null value.